packages/x-components/src/components/custom-teleport.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+  <Teleport :to="target">
+    <div ref="teleportContainer" class="x-teleport-container" :class="containerClasses">
+      <slot></slot>
+    </div>
+  </Teleport>
+</template>
+
+<script lang="ts">
+  import { defineComponent, ref, watch, onMounted, computed } from 'vue';
+
+  export default defineComponent({
+    name: 'CustomTeleport',
+    props: {
+      target: {
+        type: String,
+        required: true
+      },
+      hideTeleportContent: {
+        type: Boolean,
+        default: false
+      }
+    },
+    setup(props) {
+      const teleportContainer = ref<HTMLElement | null>(null);
+      const targetElement = ref<HTMLElement | null>(null);
+
+      const containerClasses = computed(() => ({
+        'x-teleport-container--hidden': props.hideTeleportContent
+      }));
+
+      const updateVisibility = () => {
+        if (!targetElement.value) {
+          return;
+        }
+
+        if (!props.hideTeleportContent) {
+          targetElement.value.classList.add('x-teleport-parent-container');
+        } else {
+          targetElement.value.classList.remove('x-teleport-parent-container');
+        }
```

Comment:
```suggestion
        if (!props.hideTeleportContent) {
          targetElement.value?.classList.add('x-teleport-parent-container');
        } else {
          targetElement.value?.classList.remove('x-teleport-parent-container');
        }
```

and remove previous `if (!targetElement.value) {`